pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -4385,12 +4393,23 @@ def prepare_plots(self):
                 textstr = ''
                 T2_star_str = ''
 
-                try:
-                    xunit = self.metadata["sweep_unit"]
-                    xlabel = self.metadata["sweep_name"]
-                except KeyError:
-                    xlabel = self.raw_data_dict['sweep_parameter_names'][0]
-                    xunit = self.raw_data_dict['sweep_parameter_units'][0]
+                # get x info (try for old sequences which do not have info in meta
```

Comment:
It seems that this code is replicated from prepare_projected_data_plot in MultiQubit_TimeDomain_Analysis, and it seems that prepare_projected_data_plot is called from inside prepare_plots. Why is it necessary to replicate the code here? (Is this here for the Ramsey plot while the other one if only for the projected plot?) If you need to do this here again, you should consider moving this portion of code in the super class to a separate function, which you can then call from MultiQubit_TimeDomain_Analysis.prepare_projected_data_plot, RamseyAnalysis.prepare_plots, and QScaleAnalysis.prepare_plots.

Reply:
I will add a get_xaxis_label_unit wrapper.